### ISSUE TITLE
Dynamically reconfigure an agent(telegraf), Telegraf UI on level of Graphs UI.

### ISSUE THREAD

**Issue by tierhetze** [influxdata/telegraf#3294]
## Some plugins need more flexible configuration and start/stop mechanism

I recognize that influxdata is a very powerful solution, I implemented it in a pre-production environment in a very large global trade company, that actually springs into new market (China), so it is a sort of a respect and proud, not a critique. Take it very positively. I hope in 1-2 month to see it in production, if we fully convince our OPS.

## What I lack is following:

1) Flexible text search (suitable in log analysis)

2) Ability to easily re-configure agents in runtime.

Flexible text search - I do not "demand" from you, recognizing that this is a bit out of this architecture.

Instead , I planned to configure plugins like tail and logparser and collect metrics from predefined patterns.

But the configuration is way too static. We have proprietary configurer, that accessible by OPS. Such thing we can configure 1-ce (by OPS) and then use the produced output from InfluxDb(+Grafana) (for example).

I can imagine that an issue happens in production, we ask OPS to reconfigure the plugin, restart. That takes time and phone calls. Configuration UI is not in our hands. It is a part of telegraf deployment.

What I would like is to configure the agent from web UI (Grafana, for example, but Chronograph would do too).

That will separate this part from OPS care, providing an easy access to GRAPHS WEB admins(devs mainly in our case). 

So, in short, while it is nice to have UI on the stage of graphs producing, it is also very nice to allow plugins configuration and re-start from such UI, that will give a flexibility in time of production issue analyzing.

We really need that. I even think to write such, but have no idea what it would take.

According to your architecture:
-UI adds config data to InfluxDB
-Plugins pick up this config and restart

May be something like that...

What do you think?

Thanks.

**Comment by kaithar** [comment]
I, personally, would like to see telegraf be configurable in a similar way to influx and kapacitor, in that we can change plugin config via API

**Comment by danielnelson** [comment]
There is already some work underway to provide a plugin based configuration system, with this we can support dynamic config stores such as Consul.  More information will be added to #272 when available.  This system could be used to configure Telegraf from Grafana or Chronograf.

@kaithar This system could also provide a HTTP API for configuration, but I believe it will be better to use a shared config store, especially if you have many instances of Telegraf running in your infrastructure.

Please follow #272 for updates.
